Sprint_2

AnnaNikolenko opened the pull request.

AnnaNikolenko committed:
Jacoco files are added

AnnaNikolenko committed:
Files are rewrited

AnnaNikolenko committed:
Tests are added

AnnaNikolenko committed:
POM file is added

AnnaNikolenko committed:
Update README.md

AnnaNikolenko committed:
Update README.md

AnnaNikolenko committed:
Update README.md

AnnaNikolenko committed:
Update README.md

AnnaNikolenko committed:
Update README.md

alesyaisachenko commented on src/main/java/com/example/Lion.java:
```diff
@@ -1,32 +1,58 @@
+
+    //внедрение зависимости feline через конструктор
+    public Lion(Feline feline) {
+        this.feline = feline;
```
этот конструктор не удачный.
Переменные класса должны быть заданы или при объявлении переменных или в конструкторе при создании экземпляра класса.
При использовании этого конструктора переменная hasMane не задана -- это плохо.

alesyaisachenko commented on src/test/java/com.example/tests/CatTest.java:
```diff
@@ -0,0 +1,52 @@
+        assertEquals(expected, cat.getFood()); }
+
+    @Test
+    public void getFoodTest2() throws Exception {
```
название теста не удачное -- из названия должно быть ясно чем один тест отличается от другого.

alesyaisachenko commented on src/test/java/com.example/tests/FelineTest.java:
```diff
@@ -0,0 +1,59 @@
+
+    //метод объявлен в интерфейсе Predator,  переписан в Feline и зависим от Animal
+    @Test
+    public void EatMeatTest2() throws Exception {
```
название теста не удачное -- из названия должно быть ясно чем один тест отличается от другого.

alesyaisachenko commented on src/test/java/com.example/tests/FelineTest.java:
```diff
@@ -0,0 +1,59 @@
+        assertEquals(animal.getFood("Хищник"), feline.eatMeat()); }
+
+    //метод объявлен в интерфейсе Predator,  переписан в Feline и зависим от Animal
+    @Test
```
Здесь лучше было реализовать так:
@Test(expected = Exception.class)

alesyaisachenko commented on src/test/java/com.example/tests/LionTest.java:
```diff
@@ -0,0 +1,82 @@
+
+    // метод объявленный в классе Lion и зависимый от класа Feline
+    @Test
+    public void getKittensTest2() {
```
название теста не удачное -- из названия должно быть ясно чем один тест отличается от другого.

AnnaNikolenko committed:
Jacoco file is refresh

AnnaNikolenko committed:
Files are edited

AnnaNikolenko committed:
Files are edited

Lion.java
Variable hasMane is added into the Lion(sex, feline) constructor
Constructor Lion(Feline feline) is deleted
CatTest.java
The getFoodTest2 is renamed
FelineTest.java
The EatMeatTest2() is renamed
@test(expected = Exception.class) realization is added
The getKittensTest2()
LionTest.java
The getKittensTest2() is renamed
ParametrizedAnimalTest.java
The name argument is added
ParametrizedLionTest.java
The name argument is added

AnnaNikolenko commented:
@alesyaisachenko 
Привет!
Спасибо за развернутые комментарии
Исправила и обновила файлы

alesyaisachenko commented on src/main/java/com/example/Lion.java:
```diff
@@ -1,32 +1,43 @@
+                throw new Exception();
+            }
+            this.feline = feline;
+            this.hasMane = hasMane;
```
this.hasMane = hasMane; -- это строка лишняя.
hasMane  -- задается в блоке if-else
